src/agent/command_handler/include/command_handler.hpp

```diff
@@ -56,8 +59,18 @@ namespace command_handler
                     continue;
                 }
 
-                m_commandStore.StoreCommand(cmd.value());
                 PopCommandFromQueue();
```

Comment:
This should be after the store.

Reply:
If, when the store operation fails, we mark the command as failed, it would make no difference whether the pop is called before or after. This way, we avoid having to call it twice. Because the command must be removed from the queue, as it could fail due to a duplicate ID. If we don't remove it, the command might end up being processed indefinitely.